db/migrate/20241025045928_add_invites_link_to_sidebar.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class AddInvitesLinkToSidebar < ActiveRecord::Migration[7.1]
```

Comment:
All of this looks very complicated. I think the code could be simplified by using `DB.query_hash`, `transform_keys(&:to_sym)` where needed, and by using `existing_links.partition` to group links by segment.

BUT, wouldn't it be easier to just update positions and insert the new link? Something like this, perhaps?

```ruby
community_section_id = DB.query_single(<<~SQL).first
  SELECT id
    FROM sidebar_sections
   WHERE section_type = 0
SQL

max_position = DB.query_single(<<~SQL, section_id: community_section_id)
  SELECT MAX(ssl.position)
    FROM sidebar_urls su
         JOIN sidebar_section_links ssl ON su.id = ssl.linkable_id
   WHERE ssl.linkable_type = 'SidebarUrl'
     AND ssl.sidebar_section_id = :section_id
     AND su.segment = 0
SQL

DB.exec(<<~SQL, section_id: community_section_id, position: max_position)
  UPDATE sidebar_section_links
     SET position = position + 1
   WHERE linkable_type = 'SidebarUrl'
     AND sidebar_section_id = :section_id
     AND position > :position
SQL

# Insert "new-invite" link into `sidebar_urls` and `sidebar_section_links`
# Use `max_position + 1` as `position`
```

Reply:
Agree with Gerhard here. Taking max position and insert link + 1 feels safer than removing all links and adding them again.

Reply:
I have reworked the migration, using a lot of your suggested approach. The only thing that didn't work was incrementing the `position` by 1 due to the `UNIQUE` index on `(sidebar_section_id, user_id, position)` which would be violated because the `position` of the updated rows would overlap during the `UPDATE` statement. I've resorted to deleting the rows that need to be incremented and reinserting them with `position + 1`.